packages/server/src/automations/tests/utilities/index.ts

```diff
@@ -34,6 +34,42 @@ export async function runInProd(fn: any) {
   }
 }
 
+export async function captureAllAutomationQueueMessages(
+  f: () => Promise<unknown>
+) {
+  const messages: Job<AutomationData>[] = []
+  const queue = getQueue()
+
+  const messageListener = async (message: Job<AutomationData>) => {
+    messages.push(message)
+  }
+
+  queue.on("message", messageListener)
+  try {
+    await f()
+    // Queue messages tend to be send asynchronously in API handlers, so there's
+    // no guarantee that awaiting this function will have queued anything yet.
+    // We wait here to make sure we're queued _after_ any existing async work.
+    await helpers.wait(100)
```

Comment:
Is there anyway to check if something has been queued? Just wondering if in tests this 100ms could add up quite a bit

Reply:
Unfortunately it can (and often does) happen async, so by the time we get to here the async task that does the queueing hasn't happened yet. I agree with you that this is not ideal